beacon-chain/blockchain/core.go

```diff
@@ -558,3 +513,63 @@ func (b *BeaconChain) initCycle(cState *types.CrystallizedState, aState *types.A
 
 	return newCrystallizedState, newActiveState
 }
+
+func (b *BeaconChain) hasBlock(blockhash [32]byte) (bool, error) {
+	return b.db.Has(blockKey(blockhash))
+}
+
+// saveBlock puts the passed block into the beacon chain db.
+func (b *BeaconChain) saveBlock(block *types.Block) error {
+
+	hash, err := block.Hash()
+	if err != nil {
+		return err
+	}
+
+	key := blockKey(hash)
+	encodedState, err := block.Marshal()
+	if err != nil {
+		return err
+	}
+	return b.db.Put(key, encodedState)
+}
+
+// saveCanonicalSlotNumber saves the slot number of the canonical block.
+func (b *BeaconChain) saveCanonicalSlotNumber(slotnumber uint64, hash [32]byte) error {
+	return b.db.Put(canonicalBlockKey(slotnumber), hash[:])
+}
+
+// saveCanonical puts the passed block into the beacon chain db
+// and also saves a "latest-head" key mapping to the block in the db.
+func (b *BeaconChain) saveCanonicalBlock(block *types.Block) error {
+	if err := b.saveBlock(block); err != nil {
+		return err
+	}
+	enc, err := block.Marshal()
+	if err != nil {
+		return err
+	}
+
+	return b.db.Put(canonicalHeadLookupKey, enc)
+}
+
+// getBlock retrieves a block from the db using its slot number and hash.
```

Comment:
Where is a slot number used? I don't see how that is relevant in this method. 

Please elaborate on this comment.

Reply:
will remove the slot number , it was from when previously the function took the slotnumber as an arg